// @azure/web-pubsub/dist/commonjs/generated/models/index.js
"use strict";
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KnownWebPubSubPermission = exports.KnownWebPubSubClientType = void 0;
/** Known values of {@link WebPubSubClientType} that the service accepts. */
var KnownWebPubSubClientType;
(function (KnownWebPubSubClientType) {
    /** Default */
    KnownWebPubSubClientType["Default"] = "Default";
    /** Mqtt */
    KnownWebPubSubClientType["Mqtt"] = "MQTT";
})(KnownWebPubSubClientType || (exports.KnownWebPubSubClientType = KnownWebPubSubClientType = {}));
/** Known values of {@link WebPubSubPermission} that the service accepts. */
var KnownWebPubSubPermission;
(function (KnownWebPubSubPermission) {
    /** SendToGroup */
    KnownWebPubSubPermission["SendToGroup"] = "sendToGroup";
    /** JoinLeaveGroup */
    KnownWebPubSubPermission["JoinLeaveGroup"] = "joinLeaveGroup";
})(KnownWebPubSubPermission || (exports.KnownWebPubSubPermission = KnownWebPubSubPermission = {}));
