// @azure/web-pubsub/dist/commonjs/generated/models/mappers.js
"use strict";
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebPubSubListConnectionsInGroupNextExceptionHeaders = exports.WebPubSubAddUserToGroupExceptionHeaders = exports.WebPubSubRemoveUserFromGroupExceptionHeaders = exports.WebPubSubRemoveUserFromAllGroupsExceptionHeaders = exports.WebPubSubSendToUserExceptionHeaders = exports.WebPubSubCloseUserConnectionsExceptionHeaders = exports.WebPubSubUserExistsExceptionHeaders = exports.WebPubSubGrantPermissionExceptionHeaders = exports.WebPubSubCheckPermissionExceptionHeaders = exports.WebPubSubRevokePermissionExceptionHeaders = exports.WebPubSubAddConnectionToGroupExceptionHeaders = exports.WebPubSubRemoveConnectionFromGroupExceptionHeaders = exports.WebPubSubListConnectionsInGroupExceptionHeaders = exports.WebPubSubSendToGroupExceptionHeaders = exports.WebPubSubCloseGroupConnectionsExceptionHeaders = exports.WebPubSubGroupExistsExceptionHeaders = exports.WebPubSubRemoveConnectionFromAllGroupsExceptionHeaders = exports.WebPubSubSendToConnectionExceptionHeaders = exports.WebPubSubConnectionExistsExceptionHeaders = exports.WebPubSubCloseConnectionExceptionHeaders = exports.WebPubSubSendToAllExceptionHeaders = exports.WebPubSubRemoveConnectionsFromGroupsExceptionHeaders = exports.WebPubSubGenerateClientTokenExceptionHeaders = exports.WebPubSubCloseAllConnectionsExceptionHeaders = exports.WebPubSubAddConnectionsToGroupsExceptionHeaders = exports.WebPubSubGroupMember = exports.GroupMemberPagedValues = exports.RemoveFromGroupsRequest = exports.ClientTokenResponse = exports.InnerError = exports.ErrorDetail = exports.AddToGroupsRequest = void 0;
exports.AddToGroupsRequest = {
    type: {
        name: "Composite",
        className: "AddToGroupsRequest",
        modelProperties: {
            groups: {
                serializedName: "groups",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
            filter: {
                serializedName: "filter",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ErrorDetail = {
    type: {
        name: "Composite",
        className: "ErrorDetail",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String",
                },
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String",
                },
            },
            target: {
                serializedName: "target",
                type: {
                    name: "String",
                },
            },
            details: {
                serializedName: "details",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorDetail",
                        },
                    },
                },
            },
            inner: {
                serializedName: "inner",
                type: {
                    name: "Composite",
                    className: "InnerError",
                },
            },
        },
    },
};
exports.InnerError = {
    type: {
        name: "Composite",
        className: "InnerError",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String",
                },
            },
            inner: {
                serializedName: "inner",
                type: {
                    name: "Composite",
                    className: "InnerError",
                },
            },
        },
    },
};
exports.ClientTokenResponse = {
    type: {
        name: "Composite",
        className: "ClientTokenResponse",
        modelProperties: {
            token: {
                serializedName: "token",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.RemoveFromGroupsRequest = {
    type: {
        name: "Composite",
        className: "RemoveFromGroupsRequest",
        modelProperties: {
            groups: {
                serializedName: "groups",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
            filter: {
                serializedName: "filter",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.GroupMemberPagedValues = {
    type: {
        name: "Composite",
        className: "GroupMemberPagedValues",
        modelProperties: {
            value: {
                serializedName: "value",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "WebPubSubGroupMember",
                        },
                    },
                },
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubGroupMember = {
    type: {
        name: "Composite",
        className: "WebPubSubGroupMember",
        modelProperties: {
            connectionId: {
                serializedName: "connectionId",
                required: true,
                type: {
                    name: "String",
                },
            },
            userId: {
                serializedName: "userId",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubAddConnectionsToGroupsExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubAddConnectionsToGroupsExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubCloseAllConnectionsExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubCloseAllConnectionsExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubGenerateClientTokenExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubGenerateClientTokenExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubRemoveConnectionsFromGroupsExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubRemoveConnectionsFromGroupsExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubSendToAllExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubSendToAllExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubCloseConnectionExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubCloseConnectionExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubConnectionExistsExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubConnectionExistsExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubSendToConnectionExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubSendToConnectionExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubRemoveConnectionFromAllGroupsExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubRemoveConnectionFromAllGroupsExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubGroupExistsExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubGroupExistsExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubCloseGroupConnectionsExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubCloseGroupConnectionsExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubSendToGroupExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubSendToGroupExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubListConnectionsInGroupExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubListConnectionsInGroupExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubRemoveConnectionFromGroupExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubRemoveConnectionFromGroupExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubAddConnectionToGroupExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubAddConnectionToGroupExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubRevokePermissionExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubRevokePermissionExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubCheckPermissionExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubCheckPermissionExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubGrantPermissionExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubGrantPermissionExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubUserExistsExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubUserExistsExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubCloseUserConnectionsExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubCloseUserConnectionsExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubSendToUserExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubSendToUserExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubRemoveUserFromAllGroupsExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubRemoveUserFromAllGroupsExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubRemoveUserFromGroupExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubRemoveUserFromGroupExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubAddUserToGroupExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubAddUserToGroupExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebPubSubListConnectionsInGroupNextExceptionHeaders = {
    type: {
        name: "Composite",
        className: "WebPubSubListConnectionsInGroupNextExceptionHeaders",
        modelProperties: {
            errorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
