// @azure/web-pubsub/dist/commonjs/generated/models/parameters.js
"use strict";
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.nextLink = exports.userId1 = exports.targetName = exports.permission = exports.continuationToken = exports.top = exports.maxPageSize = exports.group = exports.connectionId = exports.messageTtlSeconds = exports.filter = exports.excludedConnections = exports.accept3 = exports.message1 = exports.contentType2 = exports.accept2 = exports.message = exports.contentType1 = exports.groupsToRemove = exports.clientType = exports.groups = exports.expirationTimeInMinutes = exports.roles = exports.userId = exports.accept1 = exports.reason = exports.excluded = exports.hub = exports.accept = exports.groupsToAdd = exports.contentType = exports.apiVersion = exports.endpoint = void 0;
const mappers_js_1 = require("../models/mappers.js");
exports.endpoint = {
    parameterPath: "endpoint",
    mapper: {
        serializedName: "endpoint",
        required: true,
        type: {
            name: "String",
        },
    },
    skipEncoding: true,
};
exports.apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2024-12-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String",
        },
    },
};
exports.contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String",
        },
    },
};
exports.groupsToAdd = {
    parameterPath: "groupsToAdd",
    mapper: mappers_js_1.AddToGroupsRequest,
};
exports.accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String",
        },
    },
};
exports.hub = {
    parameterPath: "hub",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[A-Za-z][A-Za-z0-9_`,.[\\]]{0,127}$"),
        },
        serializedName: "hub",
        required: true,
        type: {
            name: "String",
        },
    },
};
exports.excluded = {
    parameterPath: ["options", "excluded"],
    mapper: {
        serializedName: "excluded",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
exports.reason = {
    parameterPath: ["options", "reason"],
    mapper: {
        serializedName: "reason",
        type: {
            name: "String",
        },
    },
};
exports.accept1 = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json, text/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String",
        },
    },
};
exports.userId = {
    parameterPath: ["options", "userId"],
    mapper: {
        serializedName: "userId",
        type: {
            name: "String",
        },
    },
};
exports.roles = {
    parameterPath: ["options", "roles"],
    mapper: {
        serializedName: "role",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
exports.expirationTimeInMinutes = {
    parameterPath: ["options", "expirationTimeInMinutes"],
    mapper: {
        defaultValue: 60,
        constraints: {
            InclusiveMinimum: 1,
        },
        serializedName: "minutesToExpire",
        type: {
            name: "Number",
        },
    },
};
exports.groups = {
    parameterPath: ["options", "groups"],
    mapper: {
        serializedName: "group",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
exports.clientType = {
    parameterPath: ["options", "clientType"],
    mapper: {
        defaultValue: "Default",
        serializedName: "clientType",
        type: {
            name: "String",
        },
    },
};
exports.groupsToRemove = {
    parameterPath: "groupsToRemove",
    mapper: mappers_js_1.RemoveFromGroupsRequest,
};
exports.contentType1 = {
    parameterPath: "contentType",
    mapper: {
        serializedName: "Content-Type",
        required: true,
        type: {
            name: "Enum",
            allowedValues: ["application/json", "application/octet-stream"],
        },
    },
};
exports.message = {
    parameterPath: "message",
    mapper: {
        serializedName: "message",
        required: true,
        type: {
            name: "Stream",
        },
    },
};
exports.accept2 = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String",
        },
    },
};
exports.contentType2 = {
    parameterPath: "contentType",
    mapper: {
        defaultValue: "text/plain",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String",
        },
    },
};
exports.message1 = {
    parameterPath: "message",
    mapper: {
        serializedName: "message",
        required: true,
        type: {
            name: "String",
        },
    },
};
exports.accept3 = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String",
        },
    },
};
exports.excludedConnections = {
    parameterPath: ["options", "excludedConnections"],
    mapper: {
        serializedName: "excluded",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
exports.filter = {
    parameterPath: ["options", "filter"],
    mapper: {
        serializedName: "filter",
        type: {
            name: "String",
        },
    },
};
exports.messageTtlSeconds = {
    parameterPath: ["options", "messageTtlSeconds"],
    mapper: {
        constraints: {
            InclusiveMaximum: 300,
            InclusiveMinimum: 0,
        },
        serializedName: "messageTtlSeconds",
        type: {
            name: "Number",
        },
    },
};
exports.connectionId = {
    parameterPath: "connectionId",
    mapper: {
        constraints: {
            MinLength: 1,
        },
        serializedName: "connectionId",
        required: true,
        type: {
            name: "String",
        },
    },
};
exports.group = {
    parameterPath: "group",
    mapper: {
        constraints: {
            Pattern: new RegExp("^(?!\\s+$).+$"),
            MaxLength: 1024,
            MinLength: 1,
        },
        serializedName: "group",
        required: true,
        type: {
            name: "String",
        },
    },
};
exports.maxPageSize = {
    parameterPath: ["options", "maxPageSize"],
    mapper: {
        constraints: {
            InclusiveMaximum: 200,
            InclusiveMinimum: 1,
        },
        serializedName: "maxpagesize",
        type: {
            name: "Number",
        },
    },
};
exports.top = {
    parameterPath: ["options", "top"],
    mapper: {
        constraints: {
            InclusiveMaximum: 2147483647,
            InclusiveMinimum: 1,
        },
        serializedName: "top",
        type: {
            name: "Number",
        },
    },
};
exports.continuationToken = {
    parameterPath: ["options", "continuationToken"],
    mapper: {
        serializedName: "continuationToken",
        type: {
            name: "String",
        },
    },
};
exports.permission = {
    parameterPath: "permission",
    mapper: {
        serializedName: "permission",
        required: true,
        type: {
            name: "String",
        },
    },
};
exports.targetName = {
    parameterPath: ["options", "targetName"],
    mapper: {
        serializedName: "targetName",
        type: {
            name: "String",
        },
    },
};
exports.userId1 = {
    parameterPath: "userId",
    mapper: {
        constraints: {
            MinLength: 1,
        },
        serializedName: "userId",
        required: true,
        type: {
            name: "String",
        },
    },
};
exports.nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String",
        },
    },
    skipEncoding: true,
};
